src/main/java/com/cloudbees/hudson/plugins/folder/AbstractFolder.java

```diff
@@ -190,6 +190,8 @@ protected void init() {
         }
         if (icon == null) {
             icon = new StockFolderIcon();
```

Comment:
I know that `StockFolderIcon` does not use the `owner` field... but that does not mean that somebody won't change it in the future... it would seem to be better to have this always call `icon.setOwner(this);` even if *currently* that is a no-op for `StockFolderIcon`